src/modules/peek/Peek.FilePreviewer/Previewers/UnsupportedFilePreviewer/UnsupportedFilePreviewer.cs

```diff
@@ -97,139 +102,61 @@ await Dispatcher.RunOnUiThread(async () =>
-
-                    var iconBitmap = await ThumbnailHelper.GetThumbnailAsync(Item.Path, cancellationToken)
-                        ?? await ThumbnailHelper.GetIconAsync(Item.Path, cancellationToken);
-
-                    cancellationToken.ThrowIfCancellationRequested();
-
-                    isIconValid = iconBitmap != null;
-
-                    Preview.IconPreview = iconBitmap ?? new SvgImageSource(new Uri("ms-appx:///Assets/Peek/DefaultFileIcon.svg"));
-                });
-            });
-
-            return isIconValid && isTaskSuccessful;
+            Preview.IconPreview = await ThumbnailHelper.GetThumbnailAsync(Item.Path, cancellationToken) ??
+                await ThumbnailHelper.GetIconAsync(Item.Path, cancellationToken) ??
+                DefaultIcon;
         }
 
-        public async Task<bool> LoadDisplayInfoAsync(CancellationToken cancellationToken)
+        private async Task LoadDisplayInfoAsync(IProgress<string?> sizeProgress, CancellationToken cancellationToken)
         {
-            bool isDisplayValid = false;
-
-            var isTaskSuccessful = await TaskExtension.RunSafe(async () =>
-            {
-                cancellationToken.ThrowIfCancellationRequested();
-
-                var type = await Task.Run(Item.GetContentTypeAsync);
-
-                cancellationToken.ThrowIfCancellationRequested();
-
-                isDisplayValid = type != null;
-
-                var readableFileSize = string.Empty;
+            string? type = await Item.GetContentTypeAsync();
 
-                if (Item is FileItem)
-                {
-                    readableFileSize = ReadableStringHelper.BytesToReadableString(Item.FileSizeBytes);
-                }
-                else if (Item is FolderItem)
-                {
-                    ComputeFolderSize(cancellationToken);
-                }
+            Dispatcher.TryEnqueue(() => Preview.FileType = type);
 
-                await Dispatcher.RunOnUiThread(() =>
-                {
-                    Preview.FileSize = readableFileSize;
-                    Preview.FileType = type;
-                    return Task.CompletedTask;
-                });
-            });
-
-            return isDisplayValid && isTaskSuccessful;
+            if (Item is FolderItem folderItem)
+            {
+                await Task.Run(() => CalculateFolderSizeWithProgress(Item.Path, sizeProgress, cancellationToken), cancellationToken);
+            }
+            else
+            {
+                ReportProgress(sizeProgress, Item.FileSizeBytes);
+            }
         }
 
-        private bool HasFailedLoadingPreview()
+        private void CalculateFolderSizeWithProgress(string path, IProgress<string?> progress, CancellationToken cancellationToken)
         {
-            var isLoadingIconPreviewSuccessful = IconPreviewTask?.Result ?? false;
-            var isLoadingDisplayInfoSuccessful = DisplayInfoTask?.Result ?? false;
+            ulong totalFiles = 0;
```

Comment:
Is this variable needed?

Reply:
Ah, sorry! This was me getting ahead of myself; I'd made a suggestion to add a file count to the view in a separate issue. I'll remove it.